sympy/assumptions/ask.py

```diff
@@ -729,6 +729,44 @@ def is_true(self):
         """
         return Predicate('is_true')
 
+    # XXX: empty_matrix or just empty?
+    @predicate_memo
+    def empty_matrix(self):
+        """
+        A matrix having at least one dimension equal to zero.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, MatrixSymbol
+        >>> X = MatrixSymbol('X', 0, 0)
+        >>> Y = MatrixSymbol('Y', 2, 0)
+        >>> Z = MatrixSymbol('Z', 2, 2)
+        >>> ask(Q.empty_matrix(X))
+        True
+        >>> ask(Q.empty_matrix(Z*Y))
+        True
+        >>> ask(Q.empty_matrix(Z))
+        False
+        """
+        return Predicate('empty_matrix')
+
+    # XXX: is scalar_matrix predicat needed?
+    @predicate_memo
+    def scalar_matrix(self):
```

Comment:
Maybe there are interesting things you can do with the fact that any expression with scalar matrices has the same properties as if the matrices were numbers, but this is also true of diagonal matrices. So something like `Implies(Q.scalar_matrix(A) & Q.real_elements(A), Q.nonnegative_elements(A**2))` would be true (the latter fact obviously doesn't exist yet), but so would ``Implies(Q.diagonal(A) & Q.real_elements(A), Q.nonnegative_elements(A**2))`. So I think anything that you could say about scalar matrices, at least relating to assumptions on its elements, you could just as well say about diagonal matrices.